kasa/protocol.py

```diff
@@ -126,3 +134,152 @@ def decrypt(ciphertext: bytes) -> str:
+
+        url = "http://%s/app/handshake1" % self.host
+        resp = await session.post(url, data=self.clientBytes)
+        _LOGGER.debug("Got response of %d to handshake1", resp.status)
+        if resp.status != 200:
+            raise SmartDeviceException(
+                "Device responded with %d to handshake1" % resp.status
+            )
+        response = await resp.read()
+        self.serverBytes = response[0:16]
+        serverHash = response[16:]
+
+        _LOGGER.debug("Server bytes are: %s", binascii.hexlify(self.serverBytes))
+        _LOGGER.debug("Server hash is: %s", binascii.hexlify(serverHash))
+
+        # Check the response from the device
+        localHash = hashlib.sha256(self.clientBytes + self.authenticator).digest()
+
+        if localHash != serverHash:
+            _LOGGER.debug(
+                "Expected %s got %s in handshake1",
+                binascii.hexlify(localHash),
+                binascii.hexlify(serverHash),
+            )
+            raise SmartDeviceException("Server response doesn't match our challenge")
+        else:
+            _LOGGER.debug("handshake1 hashes match")
+
+        # We need to include only the TP_SESSIONID cookie - aiohttp's cookie handling
+        # adds a bogus TIMEOUT cookie
+        cookie = session.cookie_jar.filter_cookies(url).get("TP_SESSIONID")
+        session.cookie_jar.clear()
+        session.cookie_jar.update_cookies({"TP_SESSIONID": cookie}, URL(url))
+        _LOGGER.debug("Cookie is %s", cookie)
+
+        # Handshake 2 has the following payload:
+        #    sha256(serverBytes | authenticator)
+        url = "http://%s/app/handshake2" % self.host
+        payload = hashlib.sha256(self.serverBytes + self.authenticator).digest()
+        resp = await session.post(url, data=payload)
+        _LOGGER.debug("Got response of %d to handshake2", resp.status)
+        if resp.status != 200:
+            raise SmartDeviceException(
+                "Device responded with %d to handshake2" % resp.status
+            )
+
+        # Done handshaking, now we need to compute the encryption keys
+        agreedBytes = self.clientBytes + self.serverBytes + self.authenticator
+        self.encryptKey = hashlib.sha256(bytearray(b"lsk") + agreedBytes).digest()[:16]
+        self.hmacKey = hashlib.sha256(bytearray(b"ldk") + agreedBytes).digest()[:28]
+        fulliv = hashlib.sha256(bytearray(b"iv") + agreedBytes).digest()
+        self.iv = fulliv[:12]
+        self.seq = int.from_bytes(fulliv[-4:], "big", signed=True)
+        self.handshake_done = True
+
+    def __encrypt(self, plaintext: bytes, iv: bytes, seq: int) -> bytes:
+        cipher = AES.new(self.encryptKey, AES.MODE_CBC, iv)
+        ciphertext = cipher.encrypt(Padding.pad(plaintext, AES.block_size))
+        signature = hashlib.sha256(
+            self.hmacKey + seq.to_bytes(4, "big", signed=True) + ciphertext
+        ).digest()
+        return signature + ciphertext
```

Comment:
```suggestion

        return signature + ciphertext
```
Please separate exit points with an empty line for readability.